NebulaWorld/Factory/FactoryManager.cs

```diff
@@ -34,6 +34,31 @@ public static void Initialize()
             TargetPlanet = PLANET_NONE;
         }
 
+        public static void InitializePrebuildRequests()
+        {
+            //Load existing prebuilds to the dictionary so it will be ready to build
+            if (LocalPlayer.IsMasterClient)
+            {
+                using (GetPrebuildRequests(out var prebuildRequests))
+                {
+                    for (int i = 0; i < GameMain.data.factoryCount; i++)
+                    {
+                        PlanetFactory factory = GameMain.data.factories[i];
```

Comment:
Perhaps a `foreach` with either a linq `where()` or a null check would work better, and the same for below

Reply:
I was reading some optimization guides for Unity games when I was doing mobile games and at that time the `for` loop was better from the performance point of view than `foreach`. There is some extra overhead or memory handling.

Reply:
Ahh, okay - than perhaps I will leave this one to @hubastard as the resident Unity expert

Reply:
But the difference is almost 0 in this case of small arrays, so I will change it so it will be easier to read.

Reply:
I just changed for loop to foreach loops :)

Reply:
Tl:dr:

`foreach` on `T[]` js the same as `for` loop.

Lambdas are horrible for perf, especially when you capture locals or this.

Reply:
LINQ was designed for cases where data is _lazily_ evaluated. It just became common use because of functional programming, but not all LINQ methods allow you to write closure-less callbacks.

In example, `.Where()` on IO or a database might make sense, since that allocation will never cost more than the actual access to the slow resource.

Reply:
For loops are faster than foreach loops in most cases. But it is fine here. I also try to stay away of LINQ in unity. Since it is slower and of habits since it doesn't work well on iOS because it needs a JIT compiler.

Also, fun fact, in Unity because of .net 3.5 it is faster to do `++i` instead of `i++` in a for loop

Reply:
Array iteration is identical for loops and foreach loops. If the perf differs, it's because the dll was built with `mcs` - mono C# compiler - not `csc` (aka roslyn).

Reply:
List and other collections are different.

Reply:
https://sharplab.io/#v2:D4AQTAjAsAUCDMACciDCiDetE+UkALIgLIAUAlgHYAuA2gLqICGATi0wJ4CUm2u/AMwD2LAKZMAxgAtEpAG6tmbRFSXtuvGP2060QygGcArgFtRpViy4BuPjoC+dxI624nCZETJCARgCtRCTpGS04eLFcdYTFJGXlFSxVKNTDNXXS9Q1NzSxsnfhcCp3d8IgAxEQoaBhSNCPTo2SpqFUQAXkQABmtWgB5agDoAGVFKAHNqKR6AamnycPzdVH1jMws2TlpyejzI7ULcA5xivcJM1dFegBUAPlIrxAAPLgwXeyA===